core/src/main/java/org/infinispan/interceptors/ClusteringInterceptor.java

```diff
@@ -76,17 +76,19 @@ protected boolean isNeedReliableReturnValues(FlagAffectedCommand command) {
    }
 
    protected boolean needsRemoteGet(InvocationContext ctx, AbstractDataCommand command) {
-      if (command.hasFlag(Flag.CACHE_MODE_LOCAL)
+      if (!command.hasFlag(Flag.DELTA_WRITE)
+         && (command.hasFlag(Flag.CACHE_MODE_LOCAL)
```

Comment:
`CACHE_MODE_LOCAL` operations should not perform any remote invocation, even if that makes the cache inconsistent.